### PR TITLE
Create and Update Blazor Wasm App Registrations

### PR DESCRIPTION
**Pull request by zahalzel**
Adds support for creating and updating Blazor Wasm app registrations in Azure AD

### CODE REVIEW

File: src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs [hunk 1]
```diff
@@ -26,6 +31,7 @@ public class MicrosoftIdentityPlatformApplicationManager
             IConsoleLogger consoleLogger,
             string commandName)
         {
+            Debugger.Launch();
```
Comment (zahalzel) [inline src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs:34]: Will remove shortly

File: src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs [hunk 2]
```diff
@@ -187,109 +194,151 @@ public class MicrosoftIdentityPlatformApplicationManager
...
 
-                if (toolOptions.EnableIdToken.HasValue &&
-                    (toolOptions.EnableIdToken.Value != updatedApp.Web.ImplicitGrantSettings.EnableIdTokenIssuance))
+        /// <summary>
+        /// Updates application in Azure AD, returns updated app or null if no updates were made
+        /// </summary>
+        /// <param name="existingApplication"></param>
+        /// <param name="toolOptions"></param>
+        /// <returns></returns>
+        private Application? UpdateApplication(Application existingApplication, ProvisioningToolOptions toolOptions, bool isBlazorWasm = false)
```
Comment (deepchoudhery) [inline src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs:245]: Would it be possible to update both `UpdateApplication` method's summaries with a description of the flow of changes. It's quite confusing to read through with very basic comments. Not sure why we are calling certain things.

**Review by deepchoudhery — CHANGES_REQUESTED** [review]
A step-by-step description of the Blazor WASM flow (pertaining to the implementation) would be great for the PR description as well.

**Review by deepchoudhery — APPROVED** [review]
LGTM

**Pull request merged**